src/hardware/gus.cpp

```diff
@@ -656,17 +663,22 @@ void Gus::AudioCallback(const uint16_t requested_frames)
 
 void Gus::BeginPlayback()
 {
+	dac_enabled = ((register_data & 0x200) != 0);
+	irq_enabled = ((register_data & 0x400) != 0);
 	audio_channel->Enable(true);
 	if (prev_logged_voices != active_voices) {
 		LOG_MSG("GUS: Activated %u voices at %u Hz", active_voices,
 		        playback_rate);
 		prev_logged_voices = active_voices;
 	}
+	is_running = true;
 }
 
 void Gus::CheckIrq()
 {
-	if (irq_status && (mix_ctrl & 0x08))
+	const bool should_interrupt = irq_status & (irq_enabled ? 0xff : 0x9f);
+	const bool lines_enabled = mix_ctrl & 0x08;
+	if (should_interrupt && lines_enabled)
```

Comment:
This is so much more readable than the same code in upstream…

Reply:
It would be nice to use bitfields with human names for the bits they're interrogating as `constexpr`'s, but we would lose the tighter coupling with upstream's code-base such taht merges would become less obvious.

Perhaps when upstream moves to C++11, maybe we can all move to bitfields together, and improve the read-ability further.